### PR TITLE
`CompareBytesInWord` keccak hint

### LINKED ISSUE #479: `CompareBytesInWord ` hint (Keccak hint)
Following hint is missing : 

https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/cairo_keccak/keccak.cairo#L211

```
 %{ ids.n_bytes < ids.BYTES_IN_WORD %}
```

this hint is used in `_copy_inputs` functions, used in `_prepare_block` which  prepares a block for the block permutation

### PR DESCRIPTION
**Pull request by TAdev0**
Resolves #479 

Implementation is from @MaksymMalicki in https://github.com/NethermindEth/cairo-vm-go/pull/462 , just adapted hintcode and left only the hint

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_keccak.go [hunk 1]
```diff
@@ -485,3 +485,43 @@ func createBlockPermutationHinter(resolver hintReferenceResolver) (hinter.Hinter
...
+//   - `nBytes` is the value to be compared with BYTES_IN_WORD
+//
+// `newCompareBytesInWordHint` writes 1 or 0 to `ap` memory address depending on whether
+// `n_bytes` is lower than BYTES_IN_WORD or not
+func newCompareBytesInWordHint(nBytes hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "CompareBytesInWordHint",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> ids.n_bytes < ids.BYTES_IN_WORD
+			nBytesVal, err := hinter.ResolveAsUint64(vm, nBytes)
```
Comment (Sh0g0-1758) [inline pkg/hintrunner/zero/zerohint_keccak.go:501]: Should we not resolve this as a felt ? 

from cairo-lang : `n_bytes: felt`

Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_keccak.go:501]: same, we can assume it should fit in uint64, will add a comment just in case.

**Review by Sh0g0-1758 — APPROVED** [review]
LGTM

**Comment by Sh0g0-1758** [comment]
@TAdev0 , can you resolve the conflicts in this. Then we can merge.

**Pull request merged**